svalbard/xmlbeans/src/main/java/org/n52/svalbard/decode/SweCommonDecoderV20.java

```diff
@@ -229,6 +229,8 @@ private SweAbstractDataComponent parseAbstractDataComponent(AbstractDataComponen
             dataArrayDoc.setDataArray1((DataArrayType) abstractDataComponent);
             sosDataArray.setXml(dataArrayDoc.xmlText(getXmlOptions()));
             sosAbstractDataComponent = sosDataArray;
+        } else if (abstractDataComponent == null) {
+
```

Comment:
Please add debug or trace level logging or improve that no empty if clause is required.

Reply:
Or, even better:
```java
else if (abstractDataComponent != null) {
    throw new UnsupportedDecoderXmlInputException(this, abstractDataComponent);
}
```